identity/handler_test.go

```diff
@@ -524,191 +524,233 @@ func TestHandler(t *testing.T) {
-				})
-				require.NoError(t, reg.PrivilegedIdentityPool().CreateIdentity(context.Background(), i))
-				return i, email, password
-			}
+	t.Run("case=PATCH should update metadata_admin correctly", func(t *testing.T) {
+		uuid := x.NewUUID().String()
+		i := &identity.Identity{Traits: identity.Traits(fmt.Sprintf(`{"subject":"%s"}`, uuid))}
+		require.NoError(t, reg.PrivilegedIdentityPool().CreateIdentity(context.Background(), i))
 
-			for name, ts := range map[string]*httptest.Server{"public": publicTS, "admin": adminTS} {
-				t.Run("endpoint="+name, func(t *testing.T) {
-					i, email, password := createCredentials(t)
-					values := func(v url.Values) {
-						v.Set("identifier", email)
-						v.Set("password", password)
-					}
+		for name, ts := range map[string]*httptest.Server{"public": publicTS, "admin": adminTS} {
+			t.Run("endpoint="+name, func(t *testing.T) {
+				patch := []patch{
+					{"op": "add", "path": "/metadata_admin", "value": "metadata admin"},
+				}
 
-					// verify login works initially
-					loginResponse := testhelpers.SubmitLoginForm(t, true, ts.Client(), ts, values, false, true, 200, "")
-					require.NotEmpty(t, gjson.Get(loginResponse, "session_token").String(), "expected to find a session token, found none")
+				res := send(t, ts, "PATCH", "/identities/"+i.ID.String(), http.StatusOK, &patch)
 
-					patch := []patch{
-						{"op": "replace", "path": "/metadata_public", "value": map[string]string{"role": "user"}},
-					}
+				assert.True(t, res.Get("metadata_admin").Exists(), "%s", res.Raw)
+				assert.EqualValues(t, "metadata admin", res.Get("metadata_admin").String(), "%s", res.Raw)
+			})
+		}
+	})
 
-					res := send(t, ts, "PATCH", "/identities/"+i.ID.String(), http.StatusOK, &patch)
-					assert.EqualValues(t, "user", res.Get("metadata_public.role").String(), "%s", res.Raw)
-					assert.NotEqualValues(t, i.StateChangedAt, sqlxx.NullTime(res.Get("state_changed_at").Time()), "%s", res.Raw)
+	t.Run("case=PATCH should update nested metadata_admin fields correctly", func(t *testing.T) {
+		uuid := x.NewUUID().String()
+		i := &identity.Identity{MetadataAdmin: sqlxx.NullJSONRawMessage(fmt.Sprintf(`{"id": "%s", "allowed": true}`, uuid))}
+		require.NoError(t, reg.PrivilegedIdentityPool().CreateIdentity(context.Background(), i))
 
-					loginResponse = testhelpers.SubmitLoginForm(t, true, ts.Client(), ts, values, false, true, 200, "")
-					msgs := gjson.Get(loginResponse, "ui.messages")
-					require.Empty(t, msgs.Array(), "expected to find no messages: %s", msgs.String())
-				})
-			}
-		})
+		for name, ts := range map[string]*httptest.Server{"public": publicTS, "admin": adminTS} {
+			t.Run("endpoint="+name, func(t *testing.T) {
+				patch := []patch{
+					{"op": "replace", "path": "/metadata_admin/allowed", "value": "false"},
+				}
 
-		t.Run("case=should delete a user and no longer be able to retrieve it", func(t *testing.T) {
```

Comment:
Moved a few tests around, as they weren't in the right sub-test anyway. Probably best to review without whitespace changes.